src/gui/activitywidget.cpp

```diff
@@ -400,6 +214,154 @@ void ActivityWidget::slotOpenFile(QModelIndex indx)
     }
 }
 
+// GUI: Display the notifications
+void ActivityWidget::slotBuildNotificationDisplay(const ActivityList& list)
+{
+    int newGuiLogs = 0;
+
+    QHash<QString, int> accNotified;
+
+    foreach( auto activity, list ) {
+        NotificationWidget *widget = 0;
+
+        if( _widgetForNotifId.contains(activity._id) ) {
+            widget = _widgetForNotifId[activity._id];
+        } else {
+            widget = new NotificationWidget(this);
+            connect(widget, SIGNAL(sendNotificationRequest(QString, QString, QString)),
+                    this, SLOT(slotSendNotificationRequest(QString, QString, QString)));
+            _notificationsLayout->addWidget(widget);
+            // _ui->_notifyScroll->setMinimumHeight( widget->height());
+            _ui->_notifyScroll->setSizeAdjustPolicy(QAbstractScrollArea::AdjustToContentsOnFirstShow);
+            _widgetForNotifId[activity._id] = widget;
+        }
+
+        widget->setAccountName( activity._accName );
+        widget->setActivity( activity );
+
+        // handle gui logs. In order to NOT annoy the user with every fetching of the
+        // notifications the notification id is stored in a Set. Only if an id
+        // is not in the set, it qualifies for guiLog.
+        // Important: The _guiLoggedNotifications set must be wiped regularly which
+        // will repeat the gui log.
+
+        // after one hour, clear the gui log notification store
```

Comment:
Why is this a good choice? How can we be sure that the ids will not reappear after one hour?

Also, pls move the magic values out of the code flow.